### PR TITLE
[Opt](scanner-scheduler) Opt scanner scheduler starvation issue.

### PR DESCRIPTION
**Pull request by kaka11chen**
### Issue
When a scanner scheduler is stuck in executing a scan task, other scan tasks will starve and have no chance to execute, which will affect other queries. Currently, the scan task hopes to scan as much data as possible to reduce the overhead of scheduling switching. Currently, it hopes to obtain up to 10MB of data in `doris_scanner_row_bytes`. However, if a query scans a table with many rows of data, but the filtering rate is very high, the filter will eventually filter out a lot of data and will never get 10MB of data. It will keep getting and executing expression filtering, which will cause other scan tasks to starve.

### Solution
The current solution is to check `max_run_time_ms` by `MonotonicStopWatch`. After executing for a maximum of 1s,  it will yield self's task for other tasks. When the scan task executes some time-consuming tasks, it needs to slice to do it.

### CODE REVIEW

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

File: be/src/vec/exec/scan/vscanner.cpp [hunk 1]
```diff
@@ -127,9 +127,11 @@ Status VScanner::get_block(RuntimeState* state, Block* block, bool* eof) {
 
             // 2. Filter the output block finally.
             {
-                auto* timer = _local_state->_filter_timer;
-                SCOPED_TIMER(timer);
-                RETURN_IF_ERROR(_filter_output_block(block));
+                if (!block->empty()) {
```
Comment (zhiqiang-hhhh) [inline be/src/vec/exec/scan/vscanner.cpp:130]: in what situation will the block in queue be empty?

File: be/src/common/config.cpp [hunk 2]
```diff
@@ -281,6 +281,8 @@ DEFINE_mInt32(doris_scan_range_max_mb, "1024");
 DEFINE_mInt32(doris_scanner_row_num, "16384");
 // single read execute fragment row bytes
 DEFINE_mInt32(doris_scanner_row_bytes, "10485760");
+// single read execute fragment max run time millseconds
+DEFINE_mInt32(doris_scanner_max_run_time_ms, "1000");
```
Comment (yiguolei) [inline be/src/common/config.cpp:285]: change it to session variable?

**Comment by kaka11chen** [comment]
run buildall

File: be/src/common/config.cpp (hunk 2 above)
Comment (kaka11chen) [reply be/src/common/config.cpp:288]: I think it is a system configuration, similar with `doris_scanner_row_bytes` or `doris_scanner_row_num`, should not changed by session level.

File: be/src/vec/exec/scan/scanner_scheduler.cpp [hunk 3]
```diff
@@ -270,6 +272,10 @@ void ScannerScheduler::_scanner_scan(std::shared_ptr<ScannerContext> ctx,
                     eos = true;
                     break;
                 }
+                if (max_run_time_watch.elapsed_time() / 1e6 >
+                    config::doris_scanner_max_run_time_ms) {
```
[the remark this answers is not in the document]
Comment (kaka11chen) [reply be/src/vec/exec/scan/scanner_scheduler.cpp]: ok

**Review by zhiqiang-hhhh — APPROVED** [review]
LGTM

**Comment by kaka11chen** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Pull request merged**